### PR TITLE
Delete user account

### PR DESCRIPTION
**Pull request by akjackson1**

### CODE REVIEW

File: application/controllers/User.php [hunk 1]
```diff
@@ -570,6 +571,37 @@ public function save_permissions() {
 			$this->output->set_output( json_encode( [ "success" => "Permissions updated successfully" ] ) );
 		}
 	}
+	
+	public function delete() {
```
Comment (vibby) [inline application/controllers/User.php]: This action should be called through the «delete» http method

Comment (vibby) [reply application/controllers/User.php]: There is not cross site protection in the action. Someone malicious can send an url to a real admin that delete a user account without confirmation nor reversibility if he is connected.

File: application/controllers/User.php [hunk 2]
```diff
@@ -570,6 +571,37 @@ public function save_permissions() {
...
+			"skills" => null,
+			"product_notify" => null,
+			"pro_translator" => null,
+		];
+		
+		$this->db->where( "id", $this->ion_auth->user()->row()->id );
+		$this->db->update( "users", $data );
+		$this->ion_auth->logout();
+		
+		$this->output->set_output( json_encode( [ "redirect" => "/" ] ) );
```
Comment (vibby) [inline application/controllers/User.php]: could we have a confirmation message on next page ?

File: application/controllers/User.php (hunk 1 above)
Comment (akjackson1) [reply application/controllers/User.php]: Good point, I tried implementing CSRF but it got a bit complicated, so I'll create another to do item for now so we can work on that later.

File: application/controllers/User.php (hunk 2 above)
Comment (akjackson1) [reply application/controllers/User.php]: Done

**Pull request merged**